### PR TITLE
Add Terms Of Use to Sign Up

### LINKED ISSUE #228: Add terms of use to the user sign up

### PR DESCRIPTION
**Pull request by RupengLiu**
close #228
unchecked the error
image
checked the error
image
show terms of use
image

### CODE REVIEW

File: src/components/users/user-signup/ko/runtime/user-signup.ts [hunk 1]
```diff
@@ -84,10 +84,9 @@ export class UserSignup {
         }
 
         const settings = {
```
Comment (azaslonov) [inline src/components/users/user-signup/ko/runtime/user-signup.ts]: A couple of things here:
1. We need to take into account that terms of use may not be specified at all.
2. Settings `userRegistrationTermsEnabled` and `userRegistrationTermsConsentRequired` should also come from configuration.

Comment (RupengLiu) [reply src/components/users/user-signup/ko/runtime/user-signup.ts]: k~, will fix that

File: src/components/users/user-signup/ko/runtime/user-signup.ts [hunk 2]
```diff
@@ -84,20 +92,15 @@ export class UserSignup {
...
-            // TODO: Registration terms could be rendered at publish time
-            userRegistrationTerms: "Test userRegistrationTerms!!!",
-            userRegistrationTermsEnabled: false,
-            userRegistrationTermsConsentRequired: false
+            userRegistrationTerms: this.termsOfUse(),
+            userRegistrationTermsEnabled: this.termsEnabled(),
+            userRegistrationTermsConsentRequired: this.isConsentRequired()
         };
-
         this.tenantSettings = settings as TenantSettings;
```
Comment (ygrik) [inline src/components/users/user-signup/ko/runtime/user-signup.ts]: I do not see any good reasons to keep this **tenantSettings **
do you really need it?

Comment (RupengLiu) [reply src/components/users/user-signup/ko/runtime/user-signup.ts]: sure i will delete it

File: src/components/users/user-signup/ko/userSignupViewModelBinder.ts [hunk 3]
```diff
@@ -21,9 +30,14 @@ export class UserSignupViewModelBinder implements ViewModelBinder<UserSignupMode
         delegationParam[DelegationParameters.ReturnUrl] =  "/";
 
         const delegationUrl = await this.tenantService.getDelegationUrl(DelegationAction.signIn, delegationParam);
-        if (delegationUrl) {
-            viewModel.delegationConfig(JSON.stringify({ delegationUrl: delegationUrl}));
-        }
+        const termsOfService = await this.getTermsOfService();
+        const runtimeConfig = JSON.stringify({
```
Comment (ygrik) [inline src/components/users/user-signup/ko/userSignupViewModelBinder.ts]: You should not add config object if there is no settings

File: src/components/users/user-signup/ko/userSignupViewModelBinder.ts [hunk 4]
```diff
@@ -21,9 +30,14 @@ export class UserSignupViewModelBinder implements ViewModelBinder<UserSignupMode
         delegationParam[DelegationParameters.ReturnUrl] =  "/";
 
         const delegationUrl = await this.tenantService.getDelegationUrl(DelegationAction.signIn, delegationParam);
-        if (delegationUrl) {
-            viewModel.delegationConfig(JSON.stringify({ delegationUrl: delegationUrl}));
-        }
+        const termsOfService = await this.getTermsOfService();
```
Comment (ygrik) [inline src/components/users/user-signup/ko/userSignupViewModelBinder.ts]: Also please use try-catch for MAPI calls

**Pull request merged**